js/index.js

```diff
@@ -3,40 +3,66 @@
 function updateSubtotal(product) {
   console.log('Calculating subtotal, yey!');
 
-  //... your code goes here
+  let price = product.querySelector('.price span').innerText;
+
+  let quantity = product.querySelector('.quantity input').value;
+
+  let priceNum = parseFloat(price);
+
+  let quantityNum = parseFloat(quantity);
+
+  let subtotalElement = product.querySelector('.subtotal span');
+ 
+  let subtotalPrice = priceNum * quantityNum;
+
+  subtotalElement.innerText = subtotalPrice
+  
+  return subtotalPrice;
 }
 
+  // ITERATION 2
 function calculateAll() {
-  // code in the following two lines is added just for testing purposes.
-  // it runs when only iteration 1 is completed. at later point, it can be removed.
-  const singleProduct = document.querySelector('.product');
-  updateSubtotal(singleProduct);
-  // end of test
+let allProducts = document.getElementsByClassName('product');
 
-  // ITERATION 2
-  //... your code goes here
+let totalPrice = 0;
+
+for (let i = 0; i < allProducts.length; i++){
+  totalPrice += updateSubtotal(allProducts[i]);
+}
 
   // ITERATION 3
-  //... your code goes here
+let totalPriceElement = document.querySelector('#total-value span');
+
+totalPriceElement.innerText = totalPrice;
 }
 
 // ITERATION 4
-
 function removeProduct(event) {
   const target = event.currentTarget;
   console.log('The target in remove is:', target);
-  //... your code goes here
+  
+  let trElement = target.parentNode.parentNode;
+  let trElementParent = trElement.parentNode;
+
+  trElementParent.removeChild(trElement);
+
+  calculateAll();
 }
 
 // ITERATION 5
-
 function createProduct() {
   //... your code goes here
+
 }
 
 window.addEventListener('load', () => {
   const calculatePricesBtn = document.getElementById('calculate');
   calculatePricesBtn.addEventListener('click', calculateAll);
 
-  //... your code goes here
-});
+  let removeButtons = document.getElementsByClassName('btn btn-remove');
```

Comment:
eu aaaacho que se remover o `btn` fica melhor. Então ficaria `document.getElementsByClassName('btn-remove')` ou `querySelectorAll('.btn-remove')`.

Não sei se esse é o problema, mas se testar e não for podem vir falar comigo